### PR TITLE
added my score page

### PR DESCRIPTION
**Pull request by meiravanidjar**

### CODE REVIEW

File: src/main/java/UserSession.java [hunk 1]
```diff
@@ -22,6 +22,8 @@ public class UserSession implements Serializable {
     private User user;
     private String errorMessage = null;
 
+    private UserGameScore score;
```
Comment (ilyash-b) [inline src/main/java/UserSession.java:25]: Can't be here. It's per game, not per user.

File: src/main/java/UserSession.java [hunk 2]
```diff
@@ -95,6 +98,10 @@ public String signOut() {
         return "index.html?faces-redirect=true";
     }
 
+    public void scoreForGame(Class<? extends Game> gameClass){
+        this.score = user.scoreForGame(gameClass);
+    }
+
```
Comment (ilyash-b) [inline src/main/java/UserSession.java:104]: Can't work like this. It's not a property of the user. It's per game.

File: src/main/webapp/home.xhtml [hunk 3]
```diff
@@ -21,6 +22,7 @@
                 <h:form>
                     <!-- TODO: style it as a link -->
                     <h:commandButton value="#{lobby.gameName} - #{lobby.users.size()} user(s)" action="#{userSession.joinLobby(lobby)}"/>
+                    <p:commandLink value="#{lobby.gameName}- My Scores " action="#{userSession.scoreForGame(lobby.gameClass)}" oncomplete="window.open('scores.xhtml')"/>
```
Comment (ilyash-b) [inline src/main/webapp/home.xhtml:25]: Please don't set property on User just to fetch it later. Use the function that you need, wherever you need to fetch the data that you want.

File: src/main/webapp/scores.xhtml [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    <div class="center">
+        <br/>
+        <img src="Images/myScore.jpg" alt="myscore"/>
+        <div class="score">
+            <h:outputText id="winScore" value="Wins number: #{userSession.score.getWinsNum()}"/>
+            <br/>
+            <h:outputText id="gameScore" value="Games number: #{userSession.score.getGamesNum()}"/>
+            <br/>
+            <h:outputText id="rateScore" value="Rate: #{userSession.score.getRate()}"/>
+        </div>
```
Comment (ilyash-b) [inline src/main/webapp/scores.xhtml:31]: This should all be per game. There should be a loop over all supported games, right now it's only Checkers.

**Review by ilyash-b — CHANGES_REQUESTED** [review]

File: src/main/webapp/home.xhtml (hunk 3 above)
Comment (ilyash-b) [inline src/main/webapp/home.xhtml:25]: How `p:commandLink` is different from `h:commandLink`?

Comment (meiravanidjar) [reply src/main/webapp/home.xhtml]: p is for using primefaces dependency (I used this for using the oncomplete=window.open )

File: src/main/java/UserSession.java (hunk 1 above)
Comment (meiravanidjar) [reply src/main/java/UserSession.java]: ok , please review again
I removed the score from here ,and used the return value directly from the function (inside loop over the game classes)

**Pull request merged**